src/main/java/com/blackducksoftware/integration/gradle/HubBdioDeployer.java

```diff
@@ -0,0 +1,108 @@
+import com.blackducksoftware.integration.hub.rest.RestConnection;
+
+public class HubBdioDeployer {
+	private final Logger logger = LoggerFactory.getLogger(HubBdioDeployer.class);
+
+	private final PluginHelper pluginHelper;
+	private final Project project;
+	private final File output;
+	private final String hubUrl;
+	private final String hubUsername;
+	private final String hubPassword;
+	private final String hubTimeout;
+	private final String hubProxyHost;
+	private final String hubProxyPort;
+	private final String hubNoProxyHosts;
+	private final String hubProxyUsername;
+	private final String hubProxyPassword;
+
+	public HubBdioDeployer(final PluginHelper pluginHelper, final Project project, final File output,
+			final String hubUrl, final String hubUsername, final String hubPassword, final String hubTimeout,
+			final String hubProxyHost, final String hubProxyPort, final String hubNoProxyHosts,
+			final String hubProxyUsername, final String hubProxyPassword) {
+		this.pluginHelper = pluginHelper;
+		this.project = project;
+		this.output = output;
+		this.hubUrl = hubUrl;
+		this.hubUsername = hubUsername;
+		this.hubPassword = hubPassword;
+		this.hubTimeout = hubTimeout;
+		this.hubProxyHost = hubProxyHost;
+		this.hubProxyPort = hubProxyPort;
+		this.hubNoProxyHosts = hubNoProxyHosts;
+		this.hubProxyUsername = hubProxyUsername;
+		this.hubProxyPassword = hubProxyPassword;
+	}
+
+	public void deployToHub() {
+		logger.info("deploying bdio output");
+		final File file = pluginHelper.getBdioFile(output, project.getName());
+
+		final HubServerConfigBuilder builder = new HubServerConfigBuilder();
+		builder.setHubUrl(hubUrl);
+		builder.setUsername(hubUsername);
+		builder.setPassword(hubPassword);
+		builder.setTimeout(hubTimeout);
+		builder.setProxyHost(hubProxyHost);
+		builder.setProxyPort(hubProxyPort);
+		builder.setIgnoredProxyHosts(hubNoProxyHosts);
+		builder.setProxyUsername(hubProxyUsername);
+		builder.setProxyPassword(hubProxyPassword);
+
+		final ValidationResults<GlobalFieldKey, HubServerConfig> results = builder.build();
+		if (results.isSuccess()) {
+			try {
+				final HubServerConfig config = results.getConstructedObject();
+				uploadFileToHub(config);
+			} catch (final URISyntaxException e) {
+				logger.error("Hub URI invalid: " + e.getMessage());
+			}
+		} else {
+			logErrors(results);
+		}
+
+		logger.info("Deployed Black Duck I/O json: " + file.getAbsolutePath());
+	}
+
+	private void uploadFileToHub(final HubServerConfig config) throws URISyntaxException {
+		final RestConnection connection = new RestConnection(config.getHubUrl().toString());
+		final HubProxyInfo proxyInfo = config.getProxyInfo();
+		if (proxyInfo.shouldUseProxyForUrl(config.getHubUrl())) {
+			connection.setProxyProperties(proxyInfo);
+		}
+
+		// final HubIntRestService service = new HubIntRestService(connection);
+		// TODO: implement the rest call to upload the BDIO file
+	}
```

Comment:
correct the hub does not support this yet.  The maven plugin will have the code snippets to make the call that can be copied here into the Gradle version.